openlibrary/core/vendors.py

```diff
@@ -291,10 +292,8 @@ def _get_betterworldbooks_metadata(isbn):
 
     url = BETTERWORLDBOOKS_API_URL + isbn
     try:
-        req = urllib2.Request(url)
-        f = urllib2.urlopen(req)
-        response = f.read()
-        f.close()
+        r = requests.get(url)
```

Comment:
Note that  Requests doesn't throw an exception like urllib2 does, so it needs explicit status checking.

I've taken a crack at repairing it in #2894, but this was flagged as an issue over 3 weeks ago, so I would have hoped to not have to take it on myself (as part of the Python 3 work).